ggml-sycl.cpp

```diff
@@ -15435,7 +15630,7 @@ static void ggml_sycl_mul_mat(const ggml_tensor * src0, const ggml_tensor * src1
 #ifdef GGML_SYCL_FORCE_DMMV
             const bool use_mul_mat_vec_q = false;
 #else
-            const bool use_mul_mat_vec_q = min_compute_capability >= VER_4VEC && ggml_is_quantized(src0->type) && ggml_nrows(src1) == 1;
+            const bool use_mul_mat_vec_q = min_compute_capability >= VER_4VEC && ggml_is_quantized(src0->type);
```

Comment:
```suggestion
            const bool use_mul_mat_vec_q = min_compute_capability >= VER_4VEC && ggml_is_quantized(src0->type) & ggml_nrows(src1) == 1;
```
This is a breaking change for both Intel and Nvidia targets. `ggml_sycl_op_mul_mat_vec_q` asserts that `GGML_ASSERT(ggml_nrows(src1) == 1);`

Reply:
Yes actually the assertion should not be mandatory and ggml_nrows=1 wont allow the iq stages to run.

Reply:
Iv tested with the suppression of the assert. It fails on both Intel and Nvidia targets
Intel (Intel(R) Level-Zero, Intel(R) Data Center GPU Max 1100 1.3 [1.3.28454]):
There is a failure in `ggml_sycl_op_dequantize_mul_mat_vec` of 
MUL_MAT(type_a=iq2_s,type_b=f32,m=16,n=1,k=256,bs=[1,1],nr=[1,1]): GGML_ASSERT: /home/aidanbelton/source/temp/llama.cpp/ggml-sycl.cpp:14217: false


Nvidia (NVIDIA A100-PCIE-40GB 8.0 [CUDA 12.2]):
There is a failure in `ggml_sycl_op_mul_mat_q` of   MUL_MAT(type_a=iq2_s,type_b=f32,m=16,n=16,k=256,bs=[1,1],nr=[1,1]): GGML_ASSERT: /home/aidanbelton/source/temp/llama.cpp/ggml-sycl.cpp:14019: false. Issue is missing case for quantization type

Tested on PVC and A100

Reply:
Actually for both the cases, it should go to neither of the methods. It is strange that it is falling back to either dequantize_mul_mat_vec or op_mul_mat_q path. For this type only vectorised mul_mat_q should be called.